octopoes/octopoes/xtdb/query.py

```diff
@@ -146,10 +147,10 @@ def from_path(cls, path: Path) -> "Query":
 
         return query
 
-    def pull(self, ooi_type: Ref) -> "Query":
+    def pull(self, ooi_type: Ref, *, fields: str = "[*]") -> "Query":
         """By default, we pull the target type. But when using find, count, etc., you have to pull explicitly."""
 
-        self._find_clauses.append(f"(pull {self._get_object_alias(ooi_type)} [*])")
+        self._find_clauses.append(f"(pull {self._get_object_alias(ooi_type)} {fields})")
```

Comment:
Is there any hardening / cleaning of this input to avoid query injection?

Reply:
This shouldn't be a problem as long as we don't put any user controlled string in the fields. The current code only uses strings that are statically defined in the code.